src/async_fuse/memfs/cache/backend.rs

```diff
@@ -0,0 +1,491 @@
+//! The implementation of backend.
+
+use async_trait::async_trait;
+use clippy_utilities::OverflowArithmetic;
+use datenlord::config::{StorageParams, StorageS3Config};
+use futures::{stream, AsyncReadExt, AsyncWriteExt, StreamExt};
+use opendal::services::{Fs, S3};
+use opendal::{ErrorKind, Operator};
+use tracing::warn;
+
+use super::{Block, Storage};
+use crate::async_fuse::fuse::protocol::INum;
+
+/// A builder to build `BackendWrapper`.
+#[derive(Debug)]
+pub struct BackendBuilder {
+    /// The storage config
+    config: StorageParams,
+    /// The size of a block
+    block_size: usize,
+}
+
+impl BackendBuilder {
+    /// Create a backend builder.
+    #[must_use]
+    pub fn new(config: StorageParams, block_size: usize) -> Self {
+        Self { config, block_size }
+    }
+
+    /// Build the backend.
+    pub fn build(self) -> opendal::Result<BackendWrapper> {
+        let BackendBuilder { config, block_size } = self;
+
+        let operator = match config {
+            StorageParams::S3(StorageS3Config {
+                ref endpoint_url,
+                ref access_key_id,
+                ref secret_access_key,
+                ref bucket_name,
+            }) => {
+                let mut builder = S3::default();
+
+                builder
+                    .endpoint(endpoint_url)
+                    .access_key_id(access_key_id)
+                    .secret_access_key(secret_access_key)
+                    .region("auto")
+                    .bucket(bucket_name);
+
+                Operator::new(builder)?.finish()
+            }
+            StorageParams::Fs(ref root) => {
+                let root = if root.is_empty() {
```

Comment:
It’s better to make this value the default in the configuration system.

Reply:
Ok, I will do it later.